dask/dataframe/rolling.py

```diff
@@ -403,10 +402,6 @@ def apply(
         if has_keyword(meta.apply, "engine"):
             # PANDAS_GT_100
             compat_kwargs = dict(engine=engine, engine_kwargs=engine_kwargs)
-        elif engine != "cython" or engine_kwargs is not None:
-            raise NotImplementedError(
-                f"Specifying the engine requires pandas>=1.0.0. Version '{PANDAS_VERSION}' installed."
-            )
         if raw is None:
             # PANDAS_GT_100: The default changed from None to False
             raw = inspect.signature(meta.apply).parameters["raw"]
```

Comment:
Looks like the middle branch can be dropped. Not sure if there are other things needed to cleanup this code